### DISCUSSION TITLE
Automatic scheduled clusterupgrades

### DISCUSSION THREAD

**Discussion by danielchristianschroeter** [okd-project/okd#947] [Q&A]
In our environment we have 5 OKD-Clusters. Are they any easy/implemented way to install automatically new cluster upgrades?
At the moment we just installed a simple cron job (oc adm upgrade --to-latest=true) which executes every week from Monday to Saturday for every cluster.
Is it possible to install the updates in OKD with some days delay automatically? We want to install the updates out of business hours to minimize any impact.

**Comment by vrutkovs** [comment]
There is a hidden key in CVO - `--enable-auto-update` - https://github.com/openshift/cluster-version-operator/blob/master/bootstrap/bootstrap-pod.yaml#L16, which makes CVO automatically apply latest available upgrade as soon as its found. There is no way to configure time or any conditions for it however. Also, CVO most likely will rewrite itself after upgrade, so this setting needs to be restored after upgrade.

However this method seems a bit dangerous to me - you're surrendering control over the version to CVO. 
Instead I'd prefer a gitops approach - you could create an app which makes a new Gitlab/Gitea/other MR when an update is available (you could use `oc adm upgrade` or parse release-controller json to determine that). ACM would probably be a good fit to coordinate updates across clusters (most likely you'll want to update stage/testing clusters first etc.)

**Comment by vrutkovs** [comment]
Previous enhancement discussion - https://github.com/openshift/enhancements/pull/124
